src/main/java/graphql/schema/transform/FieldVisibilitySchemaTransformation.java

```diff
@@ -0,0 +1,127 @@
+package graphql.schema.transform;
+
+import static graphql.schema.SchemaTransformer.transformSchema;
+import static graphql.util.TreeTransformerUtil.deleteNode;
+
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.GraphQLSchemaElement;
+import graphql.schema.GraphQLType;
+import graphql.schema.GraphQLTypeVisitorStub;
+import graphql.schema.idl.ScalarInfo;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Transforms a schema by applying a visibility predicate to every field.
+ */
+public class FieldVisibilitySchemaTransformation {
+
+    private final VisibleFieldPredicate visibleFieldPredicate;
+
+    public FieldVisibilitySchemaTransformation(VisibleFieldPredicate visibleFieldPredicate) {
+        this.visibleFieldPredicate = visibleFieldPredicate;
+    }
+
+    /**
+     * Before and after callbacks useful for side effects (logs, stopwatches etc).
+     */
+    protected void beforeTransformation() {}
```

Comment:
don't think we need that

Reply:
I'm already using it in the gateway, for timing the transformation.

Reply:
Given that is a sequential operation, you dont need it.  Just time the outer call??

Reply:
The actual call is done done in Gateway code- it's called by Nadel. So we don't actually have a place to wrap the outer call in a stopwatch.

Reply:
K